src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -266,6 +288,22 @@ private void setJournalNodes(HashMap<String, String> journalNodes) {
     }
   }
 
+  private void setNameNodeTaskNames(HashMap<String, String> nameNodeTaskNames) {
+    try {
+      set(NAMENODE_TASKNAMES_KEY, nameNodeTaskNames);
+    } catch (Exception e) {
+      log.error("Error while setting namenodes in persistent state", e);
```

Comment:
Is logging an error the best/only reaction to an exception? Maybe retry, or propagate up?